core/domain/rights_manager.py

```diff
@@ -98,19 +101,31 @@ def validate(self):
                 'Public explorations should have no viewers specified.')
 
         owner_editor = set(self.owner_ids).intersection(set(self.editor_ids))
+        owner_translator = set(self.owner_ids).intersection(
+            set(self.translator_ids))
         owner_viewer = set(self.owner_ids).intersection(set(self.viewer_ids))
+        editor_translator = set(self.editor_ids).intersection(
+            set(self.translator_ids))
         editor_viewer = set(self.editor_ids).intersection(set(self.viewer_ids))
         if owner_editor:
             raise utils.ValidationError(
                 'A user cannot be both an owner and an editor: %s' %
                 owner_editor)
+        if owner_translator:
+            raise utils.ValidationError(
+                'A user cannot be both an owner and a translator: %s' %
+                owner_translator)
         if owner_viewer:
             raise utils.ValidationError(
                 'A user cannot be both an owner and a viewer: %s' %
                 owner_viewer)
+        if editor_translator:
```

Comment:
You forgot viewer_translator; there should be 4-choose-2 checks.

Reply:
Done!